src/core/dev_api/legacy_op_extension.hpp

```diff
@@ -9,5 +9,5 @@
 namespace ov {
 
 /** @brief Class to distinguish legacy extension. */
-class LegacyOpExtension : public BaseOpExtension {};
+class OPENVINO_API LegacyOpExtension : public BaseOpExtension {};
```

Comment:
what does it really fix?
If we really need this, I suggest adding one virtual method to this class and put its implementation to cpp. It helps to prevents issues with RTTI on clang

Reply:
We can add destructor for this class

Reply:
Destructor added

Reply:
anyway, could you please elaborate in which situation we have issues which require this fix?

Reply:
This PR fixes the issue on MacOS with clang where the test: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/extension/extension.cpp#L275 failed. The problem occurs here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/frontend.cpp#L175
The dynamic cast to `LegacyOpExtension` failed (nullptr) but dynamic cast to the `BaseOpExtension` works correctly. 
On Ubuntu with gcc everything works good.